lib/ueberauth/strategy/cognito.ex

```diff
@@ -52,17 +52,44 @@ defmodule Ueberauth.Strategy.Cognito do
   defp exchange_code_for_token(%Plug.Conn{params: %{"code" => code}} = conn) do
     http_client = Application.get_env(:ueberauth_cognito, :__http_client, :hackney)
 
-    case request_token(conn, code, http_client) do
-      {:ok, 200, _headers, client_ref} ->
-        {:ok, body} = http_client.body(client_ref)
-
-        {token, id_token} = extract_tokens(body)
+    jwt_verifier =
+      Application.get_env(
+        :ueberauth_cognito,
+        :__jwt_verifier,
+        Ueberauth.Strategy.Cognito.JwtUtilities
+      )
 
-        conn
-        |> put_private(:cognito_token, token)
-        |> put_private(:cognito_id_token, id_token)
+    %{
+      client_id: client_id,
+      aws_region: aws_region,
+      user_pool_id: user_pool_id
+    } = get_config()
 
-      _ ->
+    case request_token(conn, code, http_client) do
+      {:ok, token} ->
+        case request_jwks(http_client) do
+          {:ok, jwks} ->
+            case jwt_verifier.verify(
+                   token["id_token"],
+                   jwks,
+                   client_id,
+                   aws_region,
+                   user_pool_id
```

Comment:
Rather than passing the individual values (where it's hard to remember the order), what about passing around a Config struct?

Reply:
Done.